### PR TITLE
fixes Cropping of Main image in preview

### PR DESCRIPTION
**Pull request by keshav234156**
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `grunt jasmine`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] PR body includes `fixes #0000`-style reference to original issue #
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by keshav234156** [comment]
Screenshot from 2020-02-03 15-37-21
Screenshot from 2020-02-03 15-37-06
Screenshot from 2020-02-03 15-38-35




As you can see this very well if the width is less than col-lg (bootstrap) constraint. To deal with this we can resize the image if its width reaches the maximum level. To resize we need the base64 src of the image, but I wasn't able to get it. @VladimirMikulic  Can you please help me here.

**Comment by keshav234156** [comment]
I will also fix the test after that

**Comment by VladimirMikulic** [comment]
@keshav234156 no problem. I don't think you need base64 encoding to do this.
You are overengineering, this can be done with vanilla CSS :)

Screenshot_20200203_120245

You can just add `background-size: cover;` as an inline style to `.ple-drag-drop`.

**Comment by keshav234156** [comment]
@VladimirMikulic Ok,testing your changes

**Comment by keshav234156** [comment]
Can you please see what the outpur with this image
Screenshot from 2020-02-03 15-38-35
.

Is that the only change needed ?

**Comment by VladimirMikulic** [comment]
Looks good to me.

Screenshot_20200203_130245

Maybe you could increase the height of the image container to be 220px instead of 184px, but that would be it.

**Comment by keshav234156** [comment]
Update : -
Screenshot from 2020-02-05 02-07-32


@VladimirMikulic Can you please review it again

**Comment by keshav234156** [comment]
@VladimirMikulic You are facing this issue because you might haven't replaced https://github.com/publiclab/PublicLab.Editor/pull/396/files#diff-5b0e3367cfabe5a0173e80753b827d6cL127 this line with similar that is present in plots2 at https://github.com/publiclab/plots2/blob/master/app/views/editor/rich.html.erb#L89   .Please replace it and let me know if there are any bugs still present.Thanks!!

**Comment by keshav234156** [comment]
@VladimirMikulic Travis is failing because of this test https://github.com/publiclab/PublicLab.Editor/blob/master/spec/javascripts/main_image_module_spec.js#L25 .This is related to https://github.com/publiclab/PublicLab.Editor/pull/396/files#diff-ac897566afd6498fe749fb3b45d0d88dR26-R41 but doesn't know where the problem is occurring.
Also, I ran the test in console and output was as expected. Please help me with this.

**Comment by VladimirMikulic** [comment]
Why are you using `$("")`? You could have stuck with `onload`. That may cause the tests to fail.

**Comment by keshav234156** [comment]
@VladimirMikulic You are right!!!!Thanks a lot!!

**Comment by VladimirMikulic** [comment]
:tada: :tada: :tada:

**Comment by VladimirMikulic** [comment]
I have one question though, why are you squishing everything together and abusing spaces?

**Comment by keshav234156** [comment]
@VladimirMikulic Hmm Fixing it as well!!

**Comment by keshav234156** [comment]
@VladimirMikulic How does it look now? Please do leave a comment where ever you feel there is a readability issue or any other issue.

File: src/modules/PublicLab.MainImageModule.js [hunk 1]
```diff
@@ -21,25 +21,19 @@ module.exports = PublicLab.MainImageModule = PublicLab.Module.extend({
...
-                _module.dropEl.css('height',height_dropdown)
-                _module.dropEl.css('width',width_dropdown)
-                _module.dropEl.css('background-size',width_dropdown+'px '+height_dropdown+'px');
-
-              }
-            
-        
-          
+        if (typeof url == 'string') {
+            _module.image.onload  = function(){
```
Comment (VladimirMikulic) [inline src/modules/PublicLab.MainImageModule.js]: You have 2 spaces between `oload` and `=` :)

File: src/modules/PublicLab.MainImageModule.js [hunk 2]
```diff
@@ -21,25 +21,19 @@ module.exports = PublicLab.MainImageModule = PublicLab.Module.extend({
...
+            var height_dropdown=this.height;
```
Comment (VladimirMikulic) [inline src/modules/PublicLab.MainImageModule.js]: This should be formatted like this `var height_dropdown = this.height;`.
You can apply the same to other lines.

File: src/modules/PublicLab.MainImageModule.js [hunk 3]
```diff
@@ -21,25 +21,19 @@ module.exports = PublicLab.MainImageModule = PublicLab.Module.extend({
...
+            var width_dropdown=this.width;
+            if(this.width>700){
```
Comment (VladimirMikulic) [inline src/modules/PublicLab.MainImageModule.js]: A space missing before opening `{`.

**Comment by keshav234156** [comment]
@VladimirMikulic Some spaces where I haven't made changes are autogenerated.I don't know why

File: src/modules/PublicLab.MainImageModule.js [hunk 4]
```diff
@@ -21,25 +21,19 @@ module.exports = PublicLab.MainImageModule = PublicLab.Module.extend({
...
+              var aspect_ratio=this.width/800
+              width_dropdown=800
+              height_dropdown=this.height/aspect_ratio   
+            }
+            _module.dropEl.css('height',height_dropdown)
```
Comment (VladimirMikulic) [inline src/modules/PublicLab.MainImageModule.js]: Function arguments should be spaced.

**Comment by VladimirMikulic** [comment]
It's not that I am picky, but it's general practise in JavaScript and the squished code is really hard to read. We have Webpack for squishing and optimizations :)

**Comment by keshav234156** [comment]
Actually when in image-sequencer we had husky and eslint which took care of all these things, So I am not in the habit. We should also have it here . For now, fixing this. Thanks for your review .

**Comment by keshav234156** [comment]
@VladimirMikulic Any issues still present?

**Comment by keshav234156** [comment]
done!!Any other changes?

**Comment by VladimirMikulic** [comment]
Now it's good :)

**Comment by keshav234156** [comment]
@VladimirMikulic Thanks a lot for improving me at also for reviewing again and again so many times.

**Comment by keshav234156** [comment]
@emilyashley @jywarren  Can You please give your final review and merge this!!

**Comment by VladimirMikulic** [comment]
@keshav234156 no problem, really. Thanks for being with me the the whole time.

**Comment by keshav234156** [comment]
@jywarren Can you please merge the changes

**Comment by jywarren** [comment]
Updating branch to check this out, thank you both SO MUCH for your great work on this!

**Comment by jywarren** [comment]
This looks great -- is it possible to set a max width so that it doesn't push the file selector to a new line? 

Thank you!! Please feel free to ping me in chat about this!

**Comment by keshav234156** [comment]
@jywarren yes, by not changing https://github.com/publiclab/PublicLab.Editor/pull/396/files#diff-5b0e3367cfabe5a0173e80753b827d6cR127 and doing other necessary changes

Should I make the changes?

**Comment by keshav234156** [comment]
@jywarren  Done!!

**Comment by jywarren** [comment]
Yay!!! Great work!!! 🎉

**Pull request merged**